app/controllers/user.js

```diff
@@ -67,3 +67,20 @@ exports.signUp = (req, res, next) => {
       next(error);
     });
 };
+
+exports.listUsers = (req, res, next) => {
+  const auth = req.headers.authorization; // auth is in base64(username:password)  so we need to decode the base64
+  try {
+    const decoded = sessionsManager.decode(auth);
+  } catch (e) {
+    throw errors.unauthorizedNoLogin;
+  }
+  User.getUsers(req.query.page, req.query.limit)
+    .then(users => {
+      res.status(200).send(users);
```

Comment:
now the response body will be something like `[{ firstName: .., lastName: ..., ... }, {..}, {...}, ..]` It usually its a good practice to always responds something like `{ users: [{...}, {...}, ...] }`. The advantage of the second options is that the client will always know with type of resource you are returning and also you can add some more attributes to the response.
For example in pagination endpoints usually its very useful to send some more information about remaining pages, so the frontend can displays the number of page of elements for example.

So for example we could be sending something like:
```
{ 
  users: [...], 
  total_pages: ,
  current_page: 
}
```